assets/js/googlesitekit-idea-hub-notice.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Idea Hub Notice (for Block Editor) entrypoint.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { getItem, setItem } from './googlesitekit/api/cache';
+
+const shownNotices = [];
+
+const editorNoticeKey = ( postID ) => {
+	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
+};
+
+const loadIdeaHubNotices = async ( _global = global ) => {
+	const { wp } = _global;
+
+	const hasNotice = ( postID ) => {
+		if ( wp.data.select( 'core/notices' ).getNotices() === undefined ) {
+			return undefined;
+		}
+
+		return wp.data.select( 'core/notices' ).getNotices().some( ( notice ) => {
+			return notice.id === editorNoticeKey( postID );
+		} );
+	};
+
+	const listener = async () => {
+		// eslint-disable-next-line sitekit/acronym-case
+		const postID = wp.data.select( 'core/editor' ).getCurrentPostId();
+
+		if ( ! postID ) {
+			return;
+		}
+
+		const { cacheHit } = await getItem( editorNoticeKey( postID ) );
+
+		if ( hasNotice( postID ) === false ) {
```

Comment:
Let's return early if this condition hasn't met:
```suggestion
		if ( hasNotice( postID ) !== false ) {
			return;
		}
```